src/main/java/dev/waterdog/waterdogpe/network/downstream/SwitchDownstreamHandler.java

```diff
@@ -119,6 +119,12 @@ public final boolean handle(StartGamePacket packet) {
         injectRemoveAllPlayers(this.player.getUpstream(), playerList);
         playerList.clear();
 
+        LongSet bossbars = this.player.getBossbars();
+        for (long bossbarId : bossbars) {
+            injectRemoveBossbar(this.player.getUpstream(), bossbarId);
+        }
+        bossbars.clear();
+
```

Comment:
Changed position of this code because it needs to get executed before the bossbar entity is removed